docs/api-spec.md

```diff
@@ -48,269 +72,447 @@ There is no formal specification of the Kubernetes API and Resource Model. This
+| `script`          | string                                | REQUIRED    | Script is the contents of an executable file to execute. If Script is not empty, the Sidecar cannot have a Command or Args.      |
+| `securityContext` | [`SecurityContext`](#securitycontext) | REQUIRED    | Defines the security options the Sidecar should be run with.                                                                     |
 
-\* If `lastTransitionTime` is populated by the implementation, it must be in [RFC3339](https://tools.ietf.org/html/rfc3339).
+### SecurityContext
 
-Additionally, the resource's `status.conditions` field MUST be managed as follows to enable clients to present useful diagnostic and error information to the user.
+All other fields derived from [core.v1/SecurityContext](https://pkg.go.dev/k8s.io/api/core/v1#SecurityContext) are OPTIONAL for the purposes of this spec.
 
-If a resource describes that it must report a Condition of the `type` `Succeeded`, then it must report it in the following manner:
+| Field        | Type | Requirement | Notes                                                 |
+| ------------ | ---- | ----------- | ---------------------------------------------------- |
+| `privileged` | bool | REQUIRED    | Run the sidecar in privileged mode. Default to false |
 
-* If the `status` field is `"True"`, that means the execution finished successfully.
-* If the `status` field is `"False"`, that means the execution finished unsuccessfully -- the Condition's `reason` and `message` MUST include further diagnostic information.
-* If the `status` field is `"Unknown"`, that means the execution is still ongoing, and clients can check again later until the Condition's `status` reports either `"True"` or `"False"`.
+### TaskResult
 
-Resources MAY report Conditions with other `type`s, but none are REQUIRED or RECOMMENDED.
+Defines a result produced by a Task
 
-## Listing Resources
+| Field         | Type                          | Requirement | Notes                                                                                                                                                                                    |
+| ------------- | ----------------------------- | ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `name`        | string                        | REQUIRED    | Declares the name by which a parameter is referenced.                                                                                                                                    |
+| `type`        | [`ResultsType`](#resultstype) | REQUIRED    | Type is the user-specified type of the result.                                                                                                                                           |
```

Comment:
In the corresponding `type` for `Params` the list of REQUIRED values is also included. Should it be included here too?

Reply:
done!